File: frontend/src/api/queue.ts
import { apiFetch } from './client'

export interface QueueEntry {
  id: number
  magazineId: number | null
  magazineTitle: string | null
  issueId: number | null
  issueNumber: number | null
  title: string
  status: string
  protocol: string
  downloadClient: string | null
  size: number
  sizeLeft: number
  progress: number
  speed: number
  eta: number | null
  added: string | null
  errorMessage: string | null
}

export const getQueue = () =>
  apiFetch<QueueEntry[]>('/queue')

export const removeFromQueue = (id: number, opts?: { blocklist?: boolean; removeFromClient?: boolean }) => {
  const params = new URLSearchParams()
  if (opts?.blocklist) params.set('blocklist', 'true')
  if (opts?.removeFromClient) params.set('remove_from_client', 'true')
  const qs = params.toString() ? `?${params.toString()}` : ''
  return apiFetch<void>(`/queue/${id}${qs}`, { method: 'DELETE' })
}

export const bulkRemove = (ids: number[], blocklist?: boolean) =>
  apiFetch<void>('/queue/bulk', {
    method: 'DELETE',
    body: JSON.stringify({ ids, blocklist: blocklist ?? false }),
  })

export const triggerImport = (id: number) =>
  apiFetch<{ success: boolean; message?: string }>(`/queue/${id}/import`, { method: 'POST' })

File: backend/app/schemas/queue.py
"""Queue Pydantic schemas."""

from datetime import datetime

from app.schemas import CamelModel


class QueueItemResource(CamelModel):
    id: int
    magazine_id: int | None = None
    magazine_title: str | None = None
    issue_id: int | None = None
    issue_number: int | None = None
    title: str
    status: str  # queued, downloading, paused, postProcessing, completed, failed
    protocol: str
    download_client: str | None = None
    download_client_id: int | None = None
    download_id: str | None = None
    quality: str = "unknown"
    size: int = 0
    size_left: int = 0
    progress: float = 0.0
    speed: int = 0
    eta: int | None = None  # seconds
    added: datetime | None = None
    error_message: str | None = None


class QueueBulkDeleteRequest(CamelModel):
    ids: list[int]
    blocklist: bool = False
